src/Misc/PerformanceViewer/growableFloat32Array.ts

```diff
@@ -0,0 +1,87 @@
+const growthFactor = 1.5;
+const bytesPerItem = 32 / 8;
+
+/**
+ * A class acting as a growable float32array used in the performance viewer
+ */
+export class GrowableFloat32Array {
+    private _view: Float32Array;
+    private _length: number;
+
+    /**
+     * Creates a new GrowableFloat32Array with the desired capacity.
+     * @param _capacity The initial capacity you would like to set for the array.
+     */
+    constructor(private _capacity: number) {
+        const buffer = new ArrayBuffer(_capacity * bytesPerItem);
+        this._view = new Float32Array(buffer);
+        this._length = 0;
+    }
+
+    /**
+     * The number of items currently in the array.
+     */
+    public get length(): number {
+        return this._length;
+    }
+
+    /**
+     * Pushes items to the end of the array.
+     * @param item The item to push into the array.
+     */
+    public push(item: number) {
+        this._view[this._length] = item;
+        this._length++;
+        if (this._length >= this._capacity) {
+            this._growArray();
+        }
+    }
+
+    /**
+     * Gets value at index, NaN if no such index exists.
+     * @param index the index to get the value at.
+     * @returns the value at the index provided.
+     */
+    public at(index: number): number {
+        if (index < 0 || index >= this._length) {
+            return NaN;
+        }
+
+        return this._view[index];
+    }
+
+    /**
+     * Gets a view of the original array from start to end (exclusive of end).
+     * @param start starting index.
+     * @param end ending index.
+     * @returns a subarray of the original array.
+     */
+    public subarray(start: number, end: number): GrowableFloat32Array {

```

Comment:
This creates a copy you can manipulate, if intended would be great to know it explicit from name and comment. 

Indeed, in the comment it states Gets a view which I thought would not copy and if the intent is to not copy, let s only return a Float32Array not a growable copy.

Reply:
Yep you're right, found out Float32Array gives me enough properties to just simply return it!